File: examples/cpp/analyze.cpp
#include <cstdlib>
#include <iostream>

#include "sonare.h"

int main(int argc, char** argv) {
  if (argc != 2) {
    std::cerr << "usage: " << argv[0] << " INPUT_AUDIO\n";
    return EXIT_FAILURE;
  }

  const auto audio = sonare::Audio::from_file(argv[1]);
  const auto result = sonare::MusicAnalyzer(audio).analyze();
  std::cout << "BPM: " << result.bpm << "\nKey: " << result.key.to_string() << "\n";
  return EXIT_SUCCESS;
}

File: examples/python/analyze.py
#!/usr/bin/env python3
"""Print a compact JSON analysis of an audio file."""

from __future__ import annotations

import json
import sys

import libsonare


def main() -> int:
    if len(sys.argv) != 2:
        print(f"usage: {sys.argv[0]} INPUT_AUDIO", file=sys.stderr)
        return 2

    with libsonare.Audio.from_file(sys.argv[1]) as audio:
        result = audio.analyze()
    payload = {
        "bpm": result.bpm,
        "key": result.key.name,
        "chords": [chord.name for chord in result.chords],
        "sections": [section.name for section in result.sections],
    }
    print(json.dumps(payload, indent=2))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
